src/atomate2/vasp/flows/phonons.py

```diff
@@ -0,0 +1,326 @@
+            these values can also be provided manually. born_maker has to
+            be None
+            if full_born==False, it will expect Phonopy convention for symmetrically
+            inequivalent atoms
+            if fullborn==True, it will expect VASP convention with information for
+            every atom in unit cell. Please be careful when converting
+            structures within in this workflow
+        full_born: bool
+            reduced born file (only symmerically inequivalent atoms)
+            or full information on born
+        epsilon_static_manual: Matrix3D
+            The high-frequency dielectric constant
+            Instead of recomputing born charges and epsilon,
+            these values can also be provided manually. born_maker has to
+            be None
+        total_dft_energy: float
+            Instead of recomputing the energy of the bulk structure every time,
+            this value can also be provided in eV. Please make sure that
+            it fits to the cell you are generating within this workflow
+        supercell_matrix: list
+            instead of min_length, also a supercell_matrix can
+            be given, e.g. [[1.0,0.0,0.0],[0.0,1.0,0.0],[0.0,0.0,1.0]
+        """
+
+        if (
+            not self.use_primitive_standard_structure
+            and self.kpath_scheme != "seekpath"
+        ):
+            raise ValueError(
+                "You can only use other kpath schemes with the primitive standard structure"
+            )
+
+        if self.kpath_scheme not in [
+            "seekpath",
+            "hinuma",
+            "setyawan_curtarolo",
+            "latimer_munro",
+            "all",
+        ]:
+            raise ValueError("kpath scheme is not implemented")
+
+        jobs = []
+
+        if self.use_primitive_standard_structure:
+            # These structures are compatible with many
+            # of the kpath algorithms that are used for Materials Project
+            prim_job = structure_to_primitive(structure, self.symprec)
+            jobs.append(prim_job)
+            structure = prim_job.output
+        elif self.use_conventional_standard_structure:
+            # it could be beneficial to use conventional
+            # standard structures to arrive faster at supercells with right
+            # angels
+            conv_job = structure_to_conventional(structure, self.symprec)
+            jobs.append(conv_job)
+            structure = conv_job.output
+
+        # if supercell_matrix is None, supercell size will be determined
+        if supercell_matrix is not None:
+            self.supercell_matrix = None
+        else:
+            supercell_job = get_supercell_size(
+                structure,
+                self.min_length,
+                self.prefer_90_degrees,
+                **self.get_supercell_size_kwargs,
+            )
+            jobs.append(supercell_job)
+            self.supercell_matrix = supercell_job.output
```

Comment:
I think we shouldn't set `self.supercell_matrix` and instead this could just be:
```suggestion
        if supercell_matrix is None:
            supercell_job = get_supercell_size(
                structure,
                self.min_length,
                self.prefer_90_degrees,
                **self.get_supercell_size_kwargs,
            )
            jobs.append(supercell_job)
            supercell_matrix = supercell_job.output
```

And later on in this function replace `self.supercell_matrix` with just `supercell_matrix`

Reply:
I agree and it would also fix the bug that I had overlooked so far ...